src/main/java/com/autotune/utils/GenericRestApiClient.java

```diff
@@ -116,9 +123,15 @@ public void setAuthHeaderString(String authHeaderString) {
      * @return Json object which contains API response.
      * @throws IOException
      */
-    private JSONObject fetchMetricsJson(String methodType, String queryString) throws IOException {
+    private JSONObject fetchMetricsJson(String methodType, String queryString) throws IOException, NoSuchAlgorithmException, KeyStoreException, KeyManagementException {
         String jsonOutputInString = "";
-        try (CloseableHttpClient httpclient = HttpClients.createDefault()) {
+        SSLContext sslContext = SSLContexts.custom()
+                .loadTrustMaterial((chain, authType) -> true).build();
+        SSLConnectionSocketFactory sslConnectionSocketFactory =
+                new SSLConnectionSocketFactory(sslContext, new String[]
+                        {"SSLv2Hello", "SSLv3", "TLSv1","TLSv1.1", "TLSv1.2" }, null,
```

Comment:
Can you please check on the protocols list. SSLv3 - I remember had vulnerabilities and is not recommended

Reply:
thats right . removed SSLv3.